packages/connectivity/src/scp-cf/xsuaa-service.ts

```diff
@@ -50,31 +57,41 @@ export async function getClientCredentialsToken(
   service: string | Service,
   userJwt?: string | JwtPayload
 ): Promise<ClientCredentialsResponse> {
-  const resolvedService = resolveService(service);
-  const serviceCredentials = resolvedService.credentials;
-  const subdomainAndZoneId = getSubdomainAndZoneId(userJwt);
+  const fnArgument: XsuaaParameters = {
+    ...getSubdomainAndZoneId(userJwt),
+    serviceCredentials: resolveService(service).credentials
+  };
 
-  const xssecPromise: () => Promise<ClientCredentialsResponse> = () =>
+  const xssecPromise: (
+    arg: XsuaaParameters
+  ) => Promise<ClientCredentialsResponse> = arg =>
```

Comment:
```suggestion
  type foo = (arg: XsuaaParameters) => Promise<ClientCredentialsResponse>;
  const xssecPromise: foo = arg =>
```
[nth] I find that definition really hard to read because I overlooked the `=` (assumed it was a `=>`).
`foo` is only a placeholder of course... 😉

Reply:
[q] Maybe this could even use the `function` keyword instead of assigning an arrow function?
Same applies on line 115

Reply:
I agree made the same mistake - added the function